### ISSUE TITLE
[bug]: ARI "Endpoints" endpoint reports PJSIP resource state as offline when should be unknown

### ISSUE THREAD

**Issue by DavidMartynWood** [asterisk/asterisk#782]
### Severity

Trivial

### Versions

21

### Components/Modules

PJSIP, ARI

### Operating Environment

Debian

### Frequency of Occurrence

Constant

### Issue Description

When querying PJSIP resources using the ARI "Endpoints" API, if asterisk has not yet health checked the PJSIP resource its state is reported as "offline". With chan_sip endpoints is was reported as "unknown" until the first health check had been sent.
To perform this test I am recreating a container and so all Asterisk databases are cleared down before I perform the test.

The reproduce the issue, setup 2 peers. I also ran a softphone called loadtester_142857 which is not health checked by asterisk to illustrate how that behaves.
`[siptrunk](!)
type=peer
allowtransfer=no
allowsubscribe=no
allowoverlap=no
videosupport=no
disallow=all
allow=alaw
rtptimeout=20
directmedia=no
context=incoming_call
qualify=yes

[reg_test](siptrunk)
host=registrar.test.local
context=from_registrar

[trunk_test](siptrunk)
host=trunk.test.local
context=from_trunk`

Start Asterisk, and query the Endpoints API.

Initially with PJSIP I get something like this:
`[
  {
    "technology": "PJSIP",
    "resource": "reg_registrar",
    "state": "offline",
    "channel_ids": []
  },
  {
    "technology": "PJSIP",
    "resource": "loadtester_142857",
    "state": "offline",
    "channel_ids": []
  },
  {
    "technology": "PJSIP",
    "resource": "trunk_trunk",
    "state": "online",
    "channel_ids": []
  }
]`

And then after about 10 seconds I get the following:
`[
  {
    "technology": "PJSIP",
    "resource": "reg_registrar",
    "state": "online",
    "channel_ids": []
  },
  {
    "technology": "PJSIP",
    "resource": "loadtester_142857",
    "state": "offline",
    "channel_ids": []
  },
  {
    "technology": "PJSIP",
    "resource": "trunk_trunk",
    "state": "online",
    "channel_ids": []
  }
]`

Comparing this with the old behaviour from chan_sip...

With chan_sip I got this when I first queried the API:
`[
  {
    "technology": "PJSIP",
    "resource": "reg_registrar",
    "state": "unknown",
    "channel_ids": []
  },
  {
    "technology": "PJSIP",
    "resource": "loadtester_142857",
    "state": "unknown",
    "channel_ids": []
  },
  {
    "technology": "PJSIP",
    "resource": "trunk_trunk",
    "state": "online",
    "channel_ids": []
  }
]`

And after the health checks had completed chan_sip returned the following:
`[
  {
    "technology": "PJSIP",
    "resource": "reg_registrar",
    "state": "online",
    "channel_ids": []
  },
  {
    "technology": "PJSIP",
    "resource": "loadtester_142857",
    "state": "unknown",
    "channel_ids": []
  },
  {
    "technology": "PJSIP",
    "resource": "trunk_trunk",
    "state": "online",
    "channel_ids": []
  }
]`

I think PJSIP should behave the same way as chan_sip.

Our use case is that we want to avoid calling a peer if we know it is offline for optimisation purposes, but if its state is unknown we will try to call it under the assumption we have not yet health checked it.

### Asterisk Issue Guidelines

- [X] Yes, I have read the Asterisk Issue Guidelines

**Comment by gtjoseph** [comment]
Yeah it appears that the state ARI shows will either be online or offline when it should include unknown (which is what the initial state should be).  I've opened an internal issue but it's not a trivial change.

**Comment by DavidMartynWood** [comment]
I agree it's not a trivial change. I had planned to send in a PR today but it's far more involved than I expected. Will try to work around it for the moment since I think my use case is possible niche.
